src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -130,14 +156,26 @@ public bool ContainsSpan(TextSpan filterSpan)
 
         public bool ShouldInclude(Diagnostic diagnostic)
         {
-            if (FilterTreeOpt == null)
+            if (FilterFileOpt == null)
             {
                 return true;
             }
 
-            if (!diagnostic.Location.IsInSource || diagnostic.Location.SourceTree != FilterTreeOpt)
+            if (diagnostic.Location.IsInSource)
             {
-                return false;
+                if (FilterFileOpt?.SourceTree == null ||
```

Comment:
```suggestion
                if (FilterFileOpt.SourceTree == null ||
```

Unnecessary as this was checked above.